src/components/Facility/Investigations/ShowInvestigation.tsx

```diff
@@ -92,8 +92,25 @@
   });
 
   const handleValueChange = (value: any, name: string) => {
+    const keys = name.split(".");
+    // Validate keys to prevent prototype pollution - coderabbit suggested
+    if (
+      keys.some((key) =>
+        ["__proto__", "constructor", "prototype"].includes(key),
+      )
+    ) {
+      return;
+    }
+
     const changedFields = { ...state.changedFields };
-    set(changedFields, name, value);
+    let current = changedFields;
+    for (let i = 0; i < keys.length - 1; i++) {
+      const key = keys[i];
+      if (!current[key]) current[key] = {};
+      current = current[key];
+    }
+
+    current[keys[keys.length - 1]] = value;
```

Comment:
why
is current approach and code of low quality

Reply:
const handleValueChange = (value: any, name: string) => {
    const keys = name.split(".");
    // Validate keys to prevent prototype pollution - coderabbit suggested
    if (
      keys.some((key) =>
        ["__proto__", "constructor", "prototype"].includes(key),
      )
    ) {
      return;
    }

    const changedFields = { ...state.changedFields };
    let current = changedFields;
    for (let i = 0; i < keys.length - 1; i++) {
      const key = keys[i];
      if (!current[key]) current[key] = {};
      current = current[key];
    }

    current[keys[keys.length - 1]] = value;
    dispatch({ type: "set_changed_fields", changedFields });
  };
  
  so what will be final code

Reply:
give me example where such keys can be encountered

Reply:
is it ok if I go with my current implementation

Reply:
(_/)
( T_T)
(")_(")

whats this